compiler-plugin/src/main/java/io/ballerina/stdlib/persist/compiler/PersistRecordValidator.java

```diff
@@ -543,4 +567,459 @@ public void reportDiagnosticInfo(SyntaxNodeAnalysisContext ctx, Location locatio
         Utils.reportDiagnostic(ctx, location, code, message, diagnosticSeverity);
         this.noOfReportDiagnostic++;
     }
+
+    private void validFieldTypeAndRelation(RecordTypeDescriptorNode recordNode, TypeDefinitionNode typeDefinitionNode,
+                                           SyntaxNodeAnalysisContext ctx, Symbol symbol) {
+        String recordName = typeDefinitionNode.typeName().text();
+        NodeList<Node> fields = recordNode.fields();
+        String type;
+        Node node;
+        Optional<MetadataNode> metadata;
+        TypeDefinitionNode referenceRecord = null;
+        for (Node field : fields) {
+            String tableAssociationType = "";
+            String startValue = Constants.EMPTY;
+            boolean isArrayType = false;
+            boolean isUserDefinedType = false;
+            String hasRelationAnnotation = Constants.FALSE;
+            if (field instanceof RecordFieldWithDefaultValueNode) {
+                RecordFieldWithDefaultValueNode fieldNode = (RecordFieldWithDefaultValueNode) field;
+                node = fieldNode.typeName();
+                metadata = fieldNode.metadata();
+                startValue = fieldNode.expression().toSourceCode().trim();
+            } else {
+                RecordFieldNode fieldNode = (RecordFieldNode) field;
+                node = fieldNode.typeName();
+                metadata = fieldNode.metadata();
+            }
+            if (node instanceof OptionalTypeDescriptorNode) {
+                node = ((OptionalTypeDescriptorNode) node).typeDescriptor();
+            }
+            if (node instanceof ArrayTypeDescriptorNode) {
+                isArrayType = true;
+                ArrayTypeDescriptorNode arrayTypeDescriptorNode = ((ArrayTypeDescriptorNode) node);
+                node = arrayTypeDescriptorNode.memberTypeDesc();
+            }
+            if (node instanceof QualifiedNameReferenceNode) {
```

Comment:
What is this name type?

Reply:
This type will come here. 
 `time:Civil period;`

Reply:
So the types are defined in the node? Shall we fix the naming then? I was under the impression its the value